packages/ui/src/components/va-data-table/hooks/useTableScroll.ts

```diff
@@ -0,0 +1,65 @@
+import { getCurrentInstance, computed, ExtractPropTypes } from 'vue'
+
+import { useIntersectionObserver, useElementRef } from '../../../composables'
+
+export const useTableScrollProps = {
+  scrollTopMargin: { type: Number, default: 0 },
+  scrollBottomMargin: { type: Number, default: 0 },
+}
+
+export const useTableScrollEmits = ['scroll:top', 'scroll:bottom']
+
+export default function useTableScroll (
+  props: ExtractPropTypes<typeof useTableScrollProps>,
+  emit: (event: 'scroll:bottom' | 'scroll:top', ...args: any[]) => void,
+) {
+  // TODO: replace 'vNodeProps' with 'attrs' when we bump vue to ^3.2
+  // NOTE: unfortunately, this is not reactive
+  const vNodeProps = getCurrentInstance()?.vnode.props
+  const isTopTriggerListener = vNodeProps?.['onScroll:top'] !== undefined
+  const isBottomTriggerListener = vNodeProps?.['onScroll:bottom'] !== undefined
+
+  const scrollContainer = useElementRef()
+  const topTrigger = useElementRef()
+  const bottomTrigger = useElementRef()
+
+  const isObservable = computed(
+    () => (isTopTriggerListener || isBottomTriggerListener) && !!scrollContainer.value,
```

Comment:
I think it's okay to fire an event if there is no listener. It looks like you don't need this...

Firing event even might be more performant then having reactive variable :) And code much easier to read and understand.